plc4j/spi/src/main/java/org/apache/plc4x/java/spi/Plc4xNettyWrapper.java

```diff
@@ -231,19 +236,21 @@ public Duration getTimeout() {
             completionCallback.andThen(handler.getPacketConsumer()),
             handler.getOnTimeoutConsumer(),
             handler.getErrorConsumer(),
+            handler::confirmHandled,
+            handler::confirmError,
+            handler::cancel,
```

Comment:
I'm trying to grasp with above three lines, cause I believe they are needed, but I would like to fully understand why they are added. Are we missing error callback on driver side? Anyhow, would be good if you could leave a comment in this place. If you spotted a bug then it would be great to have a test which will prevent us from getting regressions.

Reply:
Here, a new `HandlerRegistration` is created, copying the original, in order to inject a `completionCallback` before calling the `packetConsumer`. As `confirmHandled()` and `cancel()` are called for this new `HandlerRegistration`, the original one never became done, so `ConversationContext.isDone()` has never returned `true`.

I'll try to add test for this.

Reply:
Its my bug then and test should belong to me! ;-)